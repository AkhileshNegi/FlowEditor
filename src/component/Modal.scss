@import "../variables.scss";
@import "../mixins.scss";

.modal {
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    color: #777;
    font-size: 12px;
    border: 5px solid rgba(0, 0, 0, .05);
    border-radius: $curvature + 5px;
}

.title {
    margin-left: 40px;
}

.errors {
    color: $error_color;
    padding: 5px 2px;
    font-size: 12px;
}

.header {
    padding: 16px;
    font-size: 20px;
    color: #fff;
    border-top-left-radius: $curvature;
    border-top-right-radius: $curvature;
}

.content {
    background: #fff;
    p {
        line-height: 19px;
    }
}

.footer {
    background: #f1f1f1;
    padding: 12px;
    height: 45px;
    border-bottom-left-radius: $curvature;
    border-bottom-right-radius: $curvature;
}

.section-left {
    box-sizing: border-box;
    width: 70%;
    display: inline-block;
}

.section-right {
    box-sizing: border-box;
    width: 30%;
    display: inline-block;
}

.buttons {
    @include centerVertically;
    padding-right: 5px;
}

.left-buttons {
    float: left;
}

.right-buttons {
    border: 0px solid red;
    text-align: right;
    overflow: auto;
}

.show_back {
    position: absolute;
    right: 24px;
    cursor: pointer;
}

.show_front {
    position: absolute;
    right: 24px;
    cursor: pointer;
}

.background {
    position: absolute;
    font-size: 80px;
    left: -35px;
    top: 20px;
    clip: rect(0px, 100px, 55px, 40px);
}

.container {
    -webkit-perspective: 800px;
    -moz-transform: perspective(800px);
    perspective: 800px;
    .side {
        position: absolute;
        transform-style: preserve-3d;
        -webkit-transform-style: preserve-3d;
        -moz-transform-style: preserve-3d;
        -ms-transform-style: preserve-3d;
        -webkit-transition: all 500ms cubic-bezier(0.6, -0.28, 0.735, 0.045);
        transition: all 500ms cubic-bezier(0.6, -0.28, 0.735, 0.045);
        min-width: 655px;
        -webkit-backface-visibility: hidden;
        -moz-backface-visibility: hidden;
        -ms-backface-visibility: hidden;
        backface-visibility: hidden;
    }
    .front {
        -webkit-transform: rotateX(0deg);
        -o-transform: rotateX(0deg);
        -moz-transform: rotateX(0deg);
        -ms-transform: rotateX(0deg);
        transform: rotateX(0deg);
    }
    .back {
        -o-transform: rotateX(-180deg);
        -moz-transform: rotateX(-180deg);
        -webkit-transform: rotateX(-180deg);
        transform: rotateX(-180deg);
    }
    &.flipped {
        .side {
            -webkit-transition: all 500ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
            transition: all 500ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
        }
        .front {
            -o-transform: rotateX(180deg);
            -moz-transform: rotateX(180deg);
            -webkit-transform: rotateX(180deg);
            -ms-transform: rotateX(180deg);
            transform: rotateX(180deg);
        }
        .back {
            -o-transform: rotateX(0deg);
            -moz-transform: rotateX(0deg);
            -webkit-transform: rotateX(0deg);
            -ms-transform: rotateX(0deg);
            transform: rotateX(0deg);
        }
    }
}
