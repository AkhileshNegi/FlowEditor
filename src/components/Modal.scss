@import "../variables.scss";

.modal {
    background: #fff;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    color: #777;
    font-size: 12px;
}

.errors {
    color: $error_color;
    padding:5px 2px;
    font-size: 12px;
}

.modal-header {
    padding: 16px;
    font-size: 20px;
    color: $light_text;
}

.modal-content {
    p {
        line-height: 19px;
    }
}

.modal-footer {
    background: #f6f6f6;
    padding: 12px;
    border-top: 1px solid #e6e6e6;

    .left {
        float:left;
    }

    .right {
        border: 0px solid red;
        text-align:right;
        overflow:auto;
    }
}
