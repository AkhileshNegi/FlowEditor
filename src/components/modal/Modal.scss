@import "~/variables.scss";

.modal {
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    color: #777;
    font-size: 12px;
    border: 5px solid rgba(0, 0, 0, .05);
    border-radius: $curvature + 5px;
}