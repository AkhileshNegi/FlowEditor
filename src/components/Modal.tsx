import * as React from 'react';
import * as Interfaces from '../interfaces';
import * as Renderer from '../components/Renderer';
import Config from '../services/Config';

var UUID  = require('uuid');
var ReactModal = require('react-modal');
var Select2 = require('react-select2-wrapper');


interface ModalProps {
    show: boolean;
    onModalOpen: any;
    onModalClose: any;
    className: string;
    title: JSX.Element;
    width?: string;

    // button options
    ok?: string;
    cancel?: string;
    tertiary?: string;
}

/**
 * A base modal for displaying messages or performing single button actions
 */
export class Modal extends React.Component<ModalProps, {}> {

    private ele: any

    constructor(props: ModalProps) {
        super(props);
    }

    render() {
        var customStyles = {
            content : {
                marginLeft: 'auto',
                marginRight: 'auto',
                marginTop: '40px',
                bottom: 'initial',
                padding: 'none',
                borderRadius: 'none',
                outline: 'none',
                width: this.props.width ? this.props.width : "700px",
                border: 'none'
            }
        }

        var rightButtons: JSX.Element[] = [];
        var leftButtons: JSX.Element[] = [];

        if (this.props.cancel) {
            rightButtons.push(<a key={Math.random()} href="javascript:void(0);" data-type="cancel" className='btn cancel grey' onClick={this.props.onModalClose}>{this.props.cancel}</a>)
        }
        
        // no matter what, we'll have a primary button
        rightButtons.push(<a key={Math.random()} href="javascript:void(0);" data-type="ok" className='btn ok' onClick={this.props.onModalClose}>{this.props.ok ? this.props.ok : 'Ok'}</a>)

        // our left most button if we have one
        if (this.props.tertiary) {
            leftButtons.push(<a key={Math.random()} href="javascript:void(0);" data-type="tertiary" className='btn tertiary' onClick={this.props.onModalClose}>{this.props.tertiary}</a>)
        }
        

        return (
            <ReactModal
                isOpen={this.props.show}
                onAfterOpen={this.props.onModalOpen}
                onRequestClose={this.props.onModalClose}
                style={customStyles}
                shouldCloseOnOverlayClick={false}
                contentLabel="blerg"
                closeTimeoutMS={200}>

                <div ref={(ele: any) => {this.ele = ele;}} className={"modal " + this.props.className}>
                    <div className="modal-header">
                        {this.props.title}
                    </div>
                    <div className="modal-content">
                        {this.props.children}
                    </div>
                    <div className="modal-footer">
                        <div className="left">
                            {leftButtons}
                        </div>
                        <div className="right">
                            {rightButtons}
                        </div>
                    </div>
                </div>                
            </ReactModal>
        )
    }
}

interface NodeModalProps {
    initial: Interfaces.NodeEditorProps;
    renderer: Renderer.Renderer;
    changeType: boolean;
}

interface NodeModalState {
    show: boolean;
    renderer: Renderer.Renderer;
    config: Interfaces.TypeConfig;
}

/**
 * A modal for editing node properties such as actions or a router
 */
export class NodeModal extends React.Component<NodeModalProps, NodeModalState> {
    
    private rendererMap: {[type:string]:Renderer.Renderer; } = {}
    private form: HTMLFormElement;

    context: Interfaces.FlowContext;
    
    static contextTypes = {
        flow: React.PropTypes.object,
        node: React.PropTypes.object
    }
    
    constructor(props: NodeModalProps) {
        super(props);

        // stick our initialized renderer in our map
        this.rendererMap[this.props.initial.type] = this.props.renderer;

        this.state = {
            show: false,
            renderer: this.getRenderer(this.props.initial.type, this.props.initial),
            config: this.getConfig(this.props.initial.type)
        }

        this.onModalClose = this.onModalClose.bind(this);
        this.onModalOpen = this.onModalOpen.bind(this);
        this.onChangeType = this.onChangeType.bind(this);
    }

    open() {
        this.setState({
            show: true,
            renderer: this.getRenderer(this.props.initial.type, this.props.initial),
            config: this.getConfig(this.props.initial.type)
        });
    }

    close() {
        this.setState({show: false});
    }

    getConfig(type: string) {
        for (let config of Config.get().typeConfigs) {
            if (type == config.type) {
                return config;
            }
        }
    }

    getRenderer (type: string, props?: Interfaces.NodeEditorProps): Renderer.Renderer {
        if (!(type in this.rendererMap)) {
            let config = this.getConfig(type);
            this.rendererMap[type] = new config.renderer(props);
        }
        return this.rendererMap[type]
    }

    onModalOpen() {

    }
    
    onModalClose(event: any) {
        if ($(event.target).data('type') == 'ok') {
            this.state.renderer.submit(this.context, this.form);
        } else {
            this.context.flow.onModalCancel();
        }

        // force a clean object form now that we are done
        delete this.rendererMap[this.props.initial.type];
        this.close();
    }

    onChangeType(event: any) {
        var type = event.target.value;
        this.setState({ 
            renderer: this.getRenderer(type, {type: type} as Interfaces.NodeEditorProps),
            config: this.getConfig(type)
        });
    }

    render() {
        var data: any = [];
        let options: Interfaces.TypeConfig[] = Config.get().typeConfigs;
        options.map((option: Interfaces.TypeConfig) => {
            data.push({id: option.type, text: option.description});
        });

        var renderer = this.state.renderer;
        var changeOptions: JSX.Element;

        if (this.props.changeType) {
            changeOptions = (
                <div>
                    <div className="header">When a contact arrives at this point in your flow</div>
                    <Select2
                        className={"change-type"}
                        value={renderer.props.type}
                        onChange={this.onChangeType}
                        data={data}
                    />
                </div>
            )
        }

        return (
            <Modal
                width="570px"
                key={'modal_' + this.props.initial.uuid}
                title={<div>{this.state.config.name}</div>}
                className={renderer.getClassName()}
                show={this.state.show}
                onModalClose={this.onModalClose}
                onModalOpen={this.onModalOpen}
                ok='Save'
                cancel='Cancel'
                >
                
                <div className="node-editor">
                    <form ref={(ele: any) => { this.form = ele; }}>
                        {changeOptions}
                        <div className="widgets">{renderer.renderForm()}</div>
                    </form>
                </div>
            </Modal>
        )
    }
}

export default Modal;