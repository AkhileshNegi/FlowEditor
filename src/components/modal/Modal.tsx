import * as React from 'react';
import * as ReactModal from 'react-modal';
import { connect } from 'react-redux';
import Button, { ButtonProps, ButtonTypes } from '~/components/button/Button';
import ConnectedTimeoutControl from '~/components/form/timeout/TimeoutControl';
import * as styles from '~/components/modal/Modal.scss';
import * as shared from '~/components/shared.scss';
import { Types } from '~/config/typeConfigs';
import { Case, FlowNode, SwitchRouter } from '~/flowTypes';
import { AppState } from '~/store';
import { renderIf } from '~/utils';

export interface ButtonSet {
    primary: ButtonProps;
    secondary?: ButtonProps;
    tertiary?: ButtonProps;
}

interface Buttons {
    leftButtons: JSX.Element[];
    rightButtons: JSX.Element[];
}

interface CustomStyles {
    content: { [cssProperty: string]: string | number };
}

export interface ModalPassedProps {
    show: boolean;
    buttons: ButtonSet;
    advanced?: JSX.Element;
    onModalOpen?: any;
    __className?: string;
    title: JSX.Element[];
    width?: string;
}

export interface ModalStoreProps {
    translating: boolean;
    originalNode: FlowNode;
    type: Types;
}

export type ModalProps = ModalPassedProps & ModalStoreProps;

interface ModalState {
    flipped: boolean;
}

// A base modal for displaying messages or performing single button actions
export class Modal extends React.Component<ModalProps, ModalState> {
    constructor(props: ModalProps) {
        super(props);

        this.state = {
            flipped: false
        };

        this.toggleFlip = this.toggleFlip.bind(this);
    }

    public toggleFlip(): void {
        this.setState({ flipped: !this.state.flipped });
    }

    public componentWillReceiveProps(nextProps: ModalProps): void {
        if (this.props.show !== nextProps.show && !nextProps.show) {
            this.setState({ flipped: false });
        }
    }

    private getButtons(): Buttons {
        const rightButtons: JSX.Element[] = [
            <Button key={0} {...this.props.buttons.secondary} type={ButtonTypes.secondary} />
        ];

        if (this.props.buttons.primary) {
            rightButtons.push(
                <Button key={1} {...this.props.buttons.primary} type={ButtonTypes.primary} />
            );
        }

        const leftButtons: JSX.Element[] = [];

        // Our left most button if we have one
        if (this.props.buttons.tertiary) {
            leftButtons.push(
                <Button key={0} {...this.props.buttons.tertiary} type={ButtonTypes.tertiary} />
            );
        }

        return {
            leftButtons,
            rightButtons
        };
    }

    private getTopStyle(): string {
        let topStyle: string = styles.container;

        if (this.state.flipped) {
            topStyle += ` ${styles.flipped}`;
        }

        return topStyle;
    }

    private mapSides(): JSX.Element[] {
        const children = React.Children.toArray(this.props.children);
        const hasAdvanced = children.length > 1;
        const { leftButtons, rightButtons } = this.getButtons();
        return children.map((child: React.ReactChild, childIdx: number) => {
            const classes = [styles.side];
            const isFrontForm = childIdx === 0;
            let title = this.props.title[childIdx];

            if (isFrontForm) {
                classes.push(styles.front);
            } else {
                title = (
                    <div>
                        <div className={`${styles.background} fe-cog`} />
                        <div style={{ marginLeft: '50px' }}>{title}</div>
                    </div>
                );
                classes.push(styles.back);
            }
            let flip: JSX.Element;
            if (hasAdvanced) {
                /** Don't show advanced settings for SwitchRouter unless we have cases to translate */
                let cases: Case[];
                if (this.props.originalNode && this.props.originalNode.router) {
                    ({ cases } = this.props.originalNode.router as SwitchRouter);
                }
                if (isFrontForm) {
                    if (cases && !cases.length) {
                        flip = null;
                    } else {
                        flip = (
                            <div className={styles.show_back} onClick={this.toggleFlip}>
                                <div className="fe-cog" />
                            </div>
                        );
                    }
                } else {
                    flip = (
                        <div className={styles.show_front} onClick={this.toggleFlip}>
                            <div className="fe-back" />
                        </div>
                    );
                }
            }

            return (
                <div key={`modal_side_${childIdx}`} className={classes.join(' ')}>
                    <div className={styles.modal}>
                        <div
                            className={`${styles.header} ${this.props.__className} ${
                                shared[`modal_side_${childIdx}`]
                            }`}
                        >
                            {flip}
                            {title}
                        </div>
                        <div className={styles.content}>{child}</div>
                        <div className={styles.footer}>
                            <div className={styles.sectionLeft}>
                                {renderIf(
                                    // prettier-ignore
                                    isFrontForm &&
                                    !this.props.translating &&
                                    this.props.type === Types.wait_for_response
                                )(<ConnectedTimeoutControl />)}
                            </div>
                            <div className={styles.sectionRight}>
                                <div className={styles.buttons}>
                                    <div className={styles.leftButtons}>{leftButtons}</div>
                                    <div className={styles.rightButtons}>{rightButtons}</div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            );
        });
    }

    public render(): JSX.Element {
        const customStyles: CustomStyles = {
            content: {
                marginLeft: 'auto',
                marginRight: 'auto',
                marginTop: '40px',
                bottom: 'initial',
                padding: 'none',
                borderRadius: 'none',
                outline: 'none',
                width: this.props.width ? this.props.width : '700px',
                border: 'none'
            }
        };
        return (
            <ReactModal
                ariaHideApp={false}
                isOpen={this.props.show}
                onAfterOpen={this.props.onModalOpen}
                onRequestClose={
                    this.props.buttons.secondary ? this.props.buttons.secondary.onClick : null
                }
                style={customStyles}
                shouldCloseOnOverlayClick={false}
                contentLabel="Modal"
            >
                <div className={this.getTopStyle()}>{this.props.children}</div>
            </ReactModal>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = ({
    flowEditor: {
        editorUI: { translating }
    },
    nodeEditor: {
        settings: { originalNode },
        typeConfig
    }
}: AppState) => ({
    // TODO: Modal should not care about flow stuff
    translating,
    originalNode,
    type: typeConfig.type
});

// To-do: type properly
const ConnectedModal = connect(
    mapStateToProps,
    {},
    null,
    {
        withRef: true
    }
)(Modal);

export default ConnectedModal;
