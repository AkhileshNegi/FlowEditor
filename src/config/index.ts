import ConfigProvider, { ConfigProviderContext } from './ConfigProvider';
import { Type, Mode, typeConfigList, getTypeConfig } from './typeConfigs';
import { Operator, operatorConfigList, getOperatorConfig } from './operatorConfigs';

export {
    ConfigProviderContext,
    Type,
    Mode,
    Operator,
    typeConfigList,
    getTypeConfig,
    operatorConfigList,
    getOperatorConfig
};

export default ConfigProvider;
