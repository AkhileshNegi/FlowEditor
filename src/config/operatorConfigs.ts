import { HIDDEN, Operator, OperatorMap, Operators, TEXT_TYPES } from '~/config/interfaces';

export const operatorConfigList: Operator[] = [
    {
        type: Operators.has_any_word,
        verboseName: 'has any of the words',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_all_words,
        verboseName: 'has all of the words',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_phrase,
        verboseName: 'has the phrase',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_only_phrase,
        verboseName: 'has only the phrase',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_beginning,
        verboseName: 'starts with',
        operands: 1
    },
    {
        type: Operators.has_text,
        verboseName: 'has some text',
        operands: 0,
        categoryName: 'Has Text',
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_number,
        verboseName: 'has a number',
        operands: 0,
        categoryName: 'Has Number'
    },
    {
        type: Operators.has_number_between,
        verboseName: 'has a number between',
        operands: 2
    },
    {
        type: Operators.has_number_lt,
        verboseName: 'has a number below',
        operands: 1
    },
    {
        type: Operators.has_number_lte,
        verboseName: 'has a number at or below',
        operands: 1
    },
    {
        type: Operators.has_number_eq,
        verboseName: 'has a number equal to',
        operands: 1
    },
    {
        type: Operators.has_number_gte,
        verboseName: 'has a number at or above',
        operands: 1
    },
    {
        type: Operators.has_number_gt,
        verboseName: 'has a number above',
        operands: 1
    },
    {
        type: Operators.has_date,
        verboseName: 'has a date',
        operands: 0,
        categoryName: 'Has Date',
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_date_lt,
        verboseName: 'has a date before',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_date_eq,
        verboseName: 'has a date equal to',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_date_gt,
        verboseName: 'has a date after',
        operands: 1,
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_run_status,
        verboseName: 'has a run status of',
        operands: 1,
        visibility: HIDDEN
    },
    {
        type: Operators.has_group,
        verboseName: 'is in the group',
        operands: 1,
        visibility: HIDDEN
    },
    {
        type: Operators.has_phone,
        verboseName: 'has a phone number',
        operands: 0,
        categoryName: 'Has Phone'
    },
    {
        type: Operators.has_email,
        verboseName: 'has an email',
        operands: 0,
        categoryName: 'Has Email',
        visibility: TEXT_TYPES
    },
    {
        type: Operators.has_error,
        verboseName: 'has an error',
        operands: 0,
        categoryName: 'Has Error',
        visibility: HIDDEN
    },
    {
        type: Operators.has_value,
        verboseName: 'is not empty',
        operands: 0,
        categoryName: 'Not Empty'
    },
    {
        type: Operators.has_wait_timed_out,
        verboseName: null,
        operands: 0,
        visibility: HIDDEN
    },
    {
        type: Operators.has_pattern,
        verboseName: 'matches regex',
        operands: 1
    }
];

export const operatorConfigMap: OperatorMap = operatorConfigList.reduce(
    (map: OperatorMap, operatorConfig: Operator) => {
        map[operatorConfig.type] = operatorConfig;
        return map;
    },
    {}
);

/**
 * Shortcut for constant lookup of operator config in operator configs map
 * @param {string} type - The type of the operator config to return, e.g. 'send_msg'
 * @returns {Object} - The operator config found at operatorConfigs[type] or -1
 */
export const getOperatorConfig = (type: Operators): Operator => operatorConfigMap[type];
