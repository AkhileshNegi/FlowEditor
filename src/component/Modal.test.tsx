import * as React from 'react';
import { Operators } from '../config/operatorConfigs';
import { Types } from '../config/typeConfigs';
import { createCase, createExit, createWaitRouterNode } from '../testUtils/assetCreators';
import { Modal, ModalProps } from './Modal';
import { composeComponentTestUtils } from '../testUtils';

const waitForRespTitle = <div key={'front'}>Wait for Response</div>;

const rulesMeta = [
    { exitUUID: 'exit-1', exitName: 'Yes', type: Operators.has_any_word, args: ['y, yes'] },
    { exitUUID: 'exit-2', exitName: 'No', type: Operators.has_any_word, args: ['n, no'] },
    {
        exitUUID: 'exit-3',
        exitName: 'No Response',
        type: Operators.has_wait_timed_out,
        args: ['@run']
    }
];

const exits = rulesMeta.map(({ exitUUID, exitName }, idx) =>
    createExit({
        uuid: exitUUID,
        name: exitName,
        destination_node_uuid: `node-${idx}`
    })
);

const cases = rulesMeta.map(({ exitUUID, type, args }, idx) =>
    createCase({
        uuid: `case-${idx}`,
        type,
        exit_uuid: exitUUID,
        args
    })
);

const nodeToEdit = createWaitRouterNode({
    exits,
    cases,
    timeout: 300
});

const baseProps: ModalProps = {
    show: true,
    buttons: {
        primary: { name: 'Save', onClick: jest.fn() },
        secondary: { name: 'Cancel', onClick: jest.fn() }
    },
    __className: 'waitForResponse',
    title: [waitForRespTitle],
    width: '600px',
    translating: false,
    nodeToEdit,
    type: Types.wait_for_response
};

const { setup, spyOn } = composeComponentTestUtils(Modal, baseProps);

describe(Modal.name, () => {
    describe('render', () => {
        it('should render timeout control', () => {
            const { wrapper } = setup(false, {}, {}, {}, {}, [<div key={1} />, <div key={2} />]);

            expect(wrapper.find('Connect(TimeoutControl)').exists()).toBeTruthy();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
