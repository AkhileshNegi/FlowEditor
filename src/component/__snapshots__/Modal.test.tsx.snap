// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render should render timeout control 1`] = `
<Modal
  __className="waitForResponse"
  buttons={
    Object {
      "primary": Object {
        "name": "Save",
        "onClick": [MockFunction],
      },
      "secondary": Object {
        "name": "Cancel",
        "onClick": [MockFunction],
      },
    }
  }
  nodeToEdit={
    Object {
      "actions": Array [],
      "exits": Array [
        Object {
          "destination_node_uuid": "node-0",
          "name": "Yes",
          "uuid": "exit-1",
        },
        Object {
          "destination_node_uuid": "node-1",
          "name": "No",
          "uuid": "exit-2",
        },
        Object {
          "destination_node_uuid": "node-2",
          "name": "No Response",
          "uuid": "exit-3",
        },
      ],
      "router": Object {
        "cases": Array [
          Object {
            "arguments": Array [
              "y, yes",
            ],
            "exit_uuid": "exit-1",
            "type": "has_any_word",
            "uuid": "case-0",
          },
          Object {
            "arguments": Array [
              "n, no",
            ],
            "exit_uuid": "exit-2",
            "type": "has_any_word",
            "uuid": "case-1",
          },
          Object {
            "arguments": Array [
              "@run",
            ],
            "exit_uuid": "exit-3",
            "type": "has_wait_timed_out",
            "uuid": "case-2",
          },
        ],
        "default_exit_uuid": null,
        "operand": "@input",
        "type": "switch",
      },
      "uuid": "wait-router",
      "wait": Object {
        "timeout": 300,
        "type": "msg",
      },
    }
  }
  show={true}
  title={
    Array [
      <div>
        Wait for Response
      </div>,
    ]
  }
  translating={false}
  type="wait_for_response"
  width="600px"
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="Modal"
    isOpen={true}
    onRequestClose={[MockFunction]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={false}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "border": "none",
          "borderRadius": "none",
          "bottom": "initial",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "40px",
          "outline": "none",
          "padding": "none",
          "width": "600px",
        },
      }
    }
  >
    <ModalPortal
      ariaHideApp={false}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Modal"
      defaultStyles={
        Object {
          "content": Object {
            "WebkitOverflowScrolling": "touch",
            "background": "#fff",
            "border": "1px solid #ccc",
            "borderRadius": "4px",
            "bottom": "40px",
            "left": "40px",
            "outline": "none",
            "overflow": "auto",
            "padding": "20px",
            "position": "absolute",
            "right": "40px",
            "top": "40px",
          },
          "overlay": Object {
            "backgroundColor": "rgba(255, 255, 255, 0.75)",
            "bottom": 0,
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
          },
        }
      }
      isOpen={true}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={false}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "border": "none",
            "borderRadius": "none",
            "bottom": "initial",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "40px",
            "outline": "none",
            "padding": "none",
            "width": "600px",
          },
        }
      }
    >
      <div
        aria-modal="true"
        className="ReactModal__Overlay ReactModal__Overlay--after-open"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.75)",
            "bottom": 0,
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          aria-label="Modal"
          className="ReactModal__Content ReactModal__Content--after-open"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "none",
              "borderRadius": "none",
              "bottom": "initial",
              "left": "40px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "none",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
              "width": "600px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="container"
          >
            <div
              className="side front"
              key="modal_side_0"
            >
              <div
                className="modal"
              >
                <div
                  className="header waitForResponse modal_side_0"
                >
                  <div
                    className="show_back"
                    onClick={[Function]}
                  >
                    <div
                      className="fe-cog"
                    />
                  </div>
                  <div
                    key="front"
                  >
                    Wait for Response
                  </div>
                </div>
                <div
                  className="content"
                >
                  <div
                    key=".$1"
                  />
                </div>
                <div
                  className="footer"
                >
                  <div
                    className="sectionLeft"
                  >
                    <Connect(TimeoutControl)>
                      <TimeoutControl
                        checked={false}
                        timeout={null}
                        updateTimeout={[Function]}
                      >
                        <div
                          className="timeoutControlContainer"
                        >
                          <div
                            className="leftSection"
                          >
                            <label
                              className="label"
                            >
                              <input
                                checked={false}
                                className="checkBox"
                                name="timeout-enabled"
                                onChange={[Function]}
                                type="checkbox"
                              />
                              Continue when there is no response
                            </label>
                          </div>
                        </div>
                      </TimeoutControl>
                    </Connect(TimeoutControl)>
                  </div>
                  <div
                    className="sectionRight"
                  >
                    <div
                      className="buttons"
                    >
                      <div
                        className="leftButtons"
                      />
                      <div
                        className="rightButtons"
                      >
                        <Button
                          key="0"
                          name="Cancel"
                          onClick={[MockFunction]}
                          type="secondary"
                        >
                          <div
                            className="btn secondary"
                            onClick={[MockFunction]}
                          >
                            Cancel
                          </div>
                        </Button>
                        <Button
                          key="1"
                          name="Save"
                          onClick={[MockFunction]}
                          type="primary"
                        >
                          <div
                            className="btn primary"
                            onClick={[MockFunction]}
                          >
                            Save
                          </div>
                        </Button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="side back"
              key="modal_side_1"
            >
              <div
                className="modal"
              >
                <div
                  className="header waitForResponse modal_side_1"
                >
                  <div
                    className="show_front"
                    onClick={[Function]}
                  >
                    <div
                      className="fe-back"
                    />
                  </div>
                  <div>
                    <div
                      className="background fe-cog"
                    />
                    <div
                      style={
                        Object {
                          "marginLeft": "50px",
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="content"
                >
                  <div
                    key=".$2"
                  />
                </div>
                <div
                  className="footer"
                >
                  <div
                    className="sectionLeft"
                  />
                  <div
                    className="sectionRight"
                  >
                    <div
                      className="buttons"
                    >
                      <div
                        className="leftButtons"
                      />
                      <div
                        className="rightButtons"
                      >
                        <Button
                          key="0"
                          name="Cancel"
                          onClick={[MockFunction]}
                          type="secondary"
                        >
                          <div
                            className="btn secondary"
                            onClick={[MockFunction]}
                          >
                            Cancel
                          </div>
                        </Button>
                        <Button
                          key="1"
                          name="Save"
                          onClick={[MockFunction]}
                          type="primary"
                        >
                          <div
                            className="btn primary"
                            onClick={[MockFunction]}
                          >
                            Save
                          </div>
                        </Button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </ModalPortal>
  </Modal>
</Modal>
`;
